clients/unshaded/src/main/java/tachyon/client/block/RemoteBlockInStream.java

```diff
@@ -30,6 +33,13 @@
 public final class RemoteBlockInStream extends BufferedBlockInStream {
   /** The address of the worker to read the data from. */
   private final InetSocketAddress mLocation;
+  /** The returned lock id after acquiring the block lock */
+  private final Long mLockId;
+
+  /** Client to communicate with the remote worker. */
+  private final WorkerClient mWorkerClient;
```

Comment:
If the caller forgot to close the stream, the lock would be leaked as well.. Is it possible to use client B to unlock a lock acquired by client A?